### PR TITLE
PR for 1.1.0 Final Release

### PR DESCRIPTION
**Pull request by apeltzer**
This adds multiple new things, e.g. Nanopore ONT assembly, hybrid assembly and polishing methods. See changelog for full details :-) 

## PR checklist
 - [x] This comment contains a description of changes (with reason)
 - [x] If you've fixed a bug or added code that should be tested, add tests!
 - [x] If necessary, also make a PR on the [nf-core/bacass branch on the nf-core/test-datasets repo]( https://github.com/nf-core/test-datasets/pull/new/nf-core/bacass)
 - [x] Ensure the test suite passes (`nextflow run . -profile test,docker`).
 - [x] Make sure your code lints (`nf-core lint .`).
 - [x] Documentation in `docs` is updated
 - [x] `CHANGELOG.md` is updated
 - [x] `README.md` is updated

### CODE REVIEW

**Comment by apeltzer** [comment]
@MaxUlysse Any idea why the branch protection test fails here?

File: docs/usage.md [hunk 1]
```diff
@@ -106,27 +110,53 @@ If `-profile` is not specified at all the pipeline will be run locally and expec
...
+```
+
+An example of properly formatted input files can be found at the [nf-core/testData](https://github.com/nf-core/test-datasets/tree/bacass). Exemplarily, this is the input used for a hybrid assembly in testing:
+
+```bash
+ID R1 R2 LongFastQ Fast5 GenomeSize
+ERR044595 https://github.com/nf-core/test-datasets/raw/bacass/ERR044595_1M_1.fastq.gz https://github.com/nf-core/test-datasets/raw/bacass/ERR044595_1M_2.fastq.gz https://github.com/nf-core/test-datasets/raw/bacass/nanopore/subset15000.fq.gz NA 2.8m
+```
+
+* `ID` The identifier to use for handling the dataset
```
Comment (jfy133) [inline docs/usage.md]:
```suggestion
* `ID` The identifier to use for handling the dataset e.g. sample name
```

Dataset is highly ambiguous in bioinf ;), so better to give an example of what you mean.

File: docs/usage.md [hunk 2]
```diff
@@ -135,27 +165,35 @@ See [Kraken2 homepage](https://ccb.jhu.edu/software/kraken2/index.shtml#download
...
 
-The format is as follows: `{'samples': {sm1: {'readunits': {ru1: {'fq1': p1, 'fq2': p2}, ...}}}, ...}`
-where sm1 is a sample name, ru1 is a readunit (fastq pair) name,
-and p1 and p2 are full paths to FastQ files 1 and 2.
-This allows to have multiple samples and multiple FastQ pairs per sample and a file-only
-configuration of pipeline behaviour.
+### `--prokka_args`
 
-This file can also list other parameters as described here (e.g. `--kraken2db`)
+This advanced option allows you to pass extra arguments to Prokka (e.g. `" --rfam"` or `" --genus name"`). For this to work you need to quote the arguments and add at least one space
```
Comment (jfy133) [inline docs/usage.md]: Please give a proper example of how this should look like. It isn't clear what you mean here by 'adding at least one space' - between the extract arguments? Before the arguments?

File: main.nf [hunk 3]
```diff
@@ -17,26 +17,30 @@ def helpMessage() {
...
+    nextflow run nf-core/bacass --input input.csv --kraken2db 'path-to-kraken2db' -profile docker
 
     Mandatory arguments:
       -profile                      Configuration profile to use. Can use multiple (comma separated)
                                     Available: conda, docker, singularity, awsbatch, test and more.
-
+      --input                       The design file used for running the pipeline.
+
+    Pipeline arguments:
+        --assembler                   Default: "Unicycler", Available: "Canu", "Miniasm", "Unicycler". Short & Hybrid assembly always runs "Unicycler".
```
Comment (jfy133) [inline main.nf]: Any reason why this section of help has an extra indent?

File: main.nf [hunk 4]
```diff
@@ -17,26 +17,30 @@ def helpMessage() {
...
-    nextflow run nf-core/bacass -params-file params.yaml -profile docker
-    or
-    nextflow run nf-core/bacass --reads '*_R{1,2}.fastq.gz' --kraken2db 'path-to-kraken2db' -profile docker
+    nextflow run nf-core/bacass --input input.csv --kraken2db 'path-to-kraken2db' -profile docker
 
     Mandatory arguments:
       -profile                      Configuration profile to use. Can use multiple (comma separated)
                                     Available: conda, docker, singularity, awsbatch, test and more.
-
+      --input                       The design file used for running the pipeline.
```
Comment (jfy133) [inline main.nf]:
```suggestion
      --input                       The design file used for running the pipeline in TSV format.
```

File: main.nf [hunk 5]
```diff
@@ -17,26 +17,30 @@ def helpMessage() {
...
-				    if not defined through `--reads` (see below). See docs for more info
-      --reads                       Path to paired-end input reads (must be surrounded with quotes; see also docs)
-                                    Can be replaced with samples dictionary in params-file (see above)
-      --skip_kraken2                Don't run Kraken2 for classification
-      --kraken2db                   Path to Kraken2 Database directory
-      --unicycler_args              Advanced: Extra arguments to Unicycler (quote and add leading space)
-      --prokka_args                 Advanced: Extra arguments to Prokka (quote and add leading space)
       --outdir                      The output directory where the results will be saved
       --email                       Set this parameter to your e-mail address to get a summary e-mail with details of the run sent to you when the workflow exits
       -name                         Name for the pipeline run. If not specified, Nextflow will automatically generate a random mnemonic.
```
Comment (jfy133) [inline main.nf]:
```suggestion
      -name                         Name for the pipeline run. If not specified, Nextflow will automatically generate a random mnemonic.
      
```

File: main.nf [hunk 6]
```diff
@@ -175,105 +209,195 @@ ${summary.collect { k,v -> "            <dt>$k</dt><dd><samp>${v ?: '<span style
...
+
+//AdapterTrimming for ONT reads
+process adapter_trimming {
+    
+    publishDir "${params.outdir}/${sample_id}/trimming/longreads/", mode: 'copy'
+
+    when: params.assembly_type == 'hybrid' || params.assembly_type == 'long'
+
+    input:
+	set sample_id, file(lr) from ch_for_long_trim
```
Comment (jfy133) [inline main.nf]: Remove indent here (github suggestion thingy is wierd and doesn't do it properly)

File: main.nf [hunk 7]
```diff
@@ -175,105 +209,195 @@ ${summary.collect { k,v -> "            <dt>$k</dt><dd><samp>${v ?: '<span style
...
+
+    output:
+    set sample_id, file('trimmed.fastq') into (ch_long_trimmed_unicycler, ch_long_trimmed_canu, ch_long_trimmed_miniasm, ch_long_trimmed_consensus, ch_long_trimmed_nanopolish, ch_long_trimmed_kraken, ch_long_trimmed_medaka)
+    file ("v_porechop.txt") into ch_porechop_version
+
+    when: !('short' in params.assembly_type)
+
+
+	script:
```
Comment (jfy133) [inline main.nf]: Another random indent

File: main.nf [hunk 8]
```diff
@@ -282,62 +406,157 @@ process unicycler {
...
+    awk '/^S/{print ">"\$2"\\n"\$3}' "${lrfastq}.gfa" | fold > assembly.fasta
+    """
+}
+
+//Run consensus for miniasm, the others don't need it.
+process consensus {
+    label 'large'
+
+    tag "$sample_id"
+	publishDir "${params.outdir}/${sample_id}/miniasm/consensus", mode: 'copy', pattern: 'assembly_consensus.fasta'
```
Comment (jfy133) [inline main.nf]: Remove extra indent

File: main.nf [hunk 9]
```diff
@@ -282,62 +406,157 @@ process unicycler {
...
+
+    input:
+    set sample_id, file(lrfastq) from ch_long_trimmed_consensus
+    file(assembly) from ch_assembly_from_miniasm
+
+    output:
+    file 'assembly_consensus.fasta' into (ch_assembly_consensus_for_nanopolish, ch_assembly_consensus_for_medaka)
+
+	script:
```
Comment (jfy133) [inline main.nf]: Remove extra indent

File: main.nf [hunk 10]
```diff
@@ -282,62 +406,157 @@ process unicycler {
...
+    output:
+    file("${sample_id}_kraken2.report")
 
-        output:
-        file("${sample_id}_kraken2.report")
+    when: !params.skip_kraken2
 
-        script:
+    script:
 	"""
```
Comment (jfy133) [inline main.nf]: This and close quotes also have extra indent

**Comment by MaxUlysse** [comment]
Yes, simple
The checkout of the repo was removed so this part: `$(git remote get-url origin) == *nf-core/bacass` don't work anymore

File: .github/workflows/branch.yml [hunk 11]
```diff
@@ -0,0 +1,15 @@
...
+on:
+  pull_request:
+    branches:
+    - master
+
+jobs:
+  test:
+    runs-on: ubuntu-18.04
+    steps:
+      # PRs are only ok if coming from an nf-core dev branch
```
Comment (MaxUlysse) [inline .github/workflows/branch.yml]:
```suggestion
       - uses: actions/checkout@v1
      # PRs are only ok if coming from an nf-core dev branch
```

Comment (MaxUlysse) [reply .github/workflows/branch.yml]: Should fix test for branch protection

**Comment by apeltzer** [comment]
Ok, all fixed now. Things should be fine and I've got two full size tests finished this night with the `dev` branch. Docs updated according to @jfy133 's suggestions so I only need a "yay" from you @MaxUlysse to get this out as an early christmas present for one of our collaborators ;_)

**Pull request merged**